packages/graphql/src/translate/utils/get-mutation-field-statements.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Node, Relationship } from "../../classes";
+import { SPATIAL_TYPES } from "../../constants";
+import mapToDbProperty from "../../utils/map-to-db-property";
+import { buildMathStatements, matchMathField, mathDescriptorBuilder } from "./math";
+import { parseMutableField } from "./parse-mutable-field";
+
+/**
+ * Transforms a key-value pair such as:
+ * { name_SET: "John" }
+ * Into statements such as:
+ * SET node.name = $param
+ **/
+export function getMutationFieldStatements({
+    nodeOrRel,
+    param,
+    key,
+    varName,
+    value,
+    withVars,
+}: {
+    nodeOrRel: Node | Relationship;
+    param: string;
+    key: string;
+    varName: string;
+    value: any;
+    withVars: string[];
+}): string {
+    const strs: string[] = [];
+    const { settableField, operator } = parseMutableField(nodeOrRel, key);
+    if (settableField) {
+        const dbFieldName = mapToDbProperty(nodeOrRel, settableField.fieldName);
+        if (settableField.typeMeta.required && value === null) {
+            throw new Error(`Cannot set non-nullable field ${nodeOrRel.name}.${settableField.fieldName} to null`);
+        }
+
+        switch (operator) {
+            case "SET": {
+                const isSpatial = SPATIAL_TYPES.includes(settableField.typeMeta.name);
+                if (isSpatial) {
+                    if (settableField.typeMeta.array) {
+                        strs.push(`SET ${varName}.${dbFieldName} = [p in $${param} | point(p)]`);
+                    } else {
+                        strs.push(`SET ${varName}.${dbFieldName} = point($${param})`);
+                    }
+                } else {
+                    strs.push(`SET ${varName}.${dbFieldName} = $${param}`);
+                }
```

Comment:
I remember doing some work so that points get put directly in the params as points so that this weird casting isn't needed... Am I imaging things?

Reply:
Ah no it was for projecting - never mind! https://github.com/neo4j/graphql/pull/5521